src/gui/library.cpp

```diff
@@ -17,10 +19,12 @@ Library::Library(stringw name, unsigned int buffer_size)
 
     for (unsigned int i = 0; i < file_list->getFileCount(); i++)
     {
-        IXMLReader* xml_reader = file_system->createXMLReader(
-            file_list->getFullFileName(i));
+        AutoDrop<IXMLReader> xml_reader(file_system->createXMLReader(
+            file_list->getFullFileName(i)));
+
+        // Make sure we don't leak element.
+        std::auto_ptr<Element> element(new Element(xml_reader.get()));
```

Comment:
Hi, please do not use `auto_ptr` as it is deprecated http://en.cppreference.com/w/cpp/memory/auto_ptr.